main/src/libraries/wrdac/src/subsystems/subSystem_KARMA.cpp

```diff
@@ -169,7 +251,145 @@ bool wysiwyd::wrdac::SubSystem_KARMA::prepare()
     return true;
 }
 
-bool wysiwyd::wrdac::SubSystem_KARMA::push(const yarp::sig::Vector &targetCenter, const double theta, const double radius, const yarp::os::Bottle &options, const std::string &sName)
+bool wysiwyd::wrdac::SubSystem_KARMA::toolAttach(const std::string &armType, const Vector &dimTool)
+{
+    yarp::os::Bottle bCmd;
+    bCmd.addVocab(yarp::os::Vocab::encode("tool"));
+    bCmd.addVocab(yarp::os::Vocab::encode("attach"));
+    appendString(bCmd,armType);
+    appendTarget(bCmd,dimTool);
+
+    bool bReturn = sendCmd(bCmd,true);
+    std::string status;
+    bReturn ? status = "success" : status = "failed";
+
+    return bReturn;
+}
+
+void wysiwyd::wrdac::SubSystem_KARMA::toolRemove()
+{
+    yarp::os::Bottle bCmd;
+    bCmd.addVocab(yarp::os::Vocab::encode("tool"));
+    bCmd.addVocab(yarp::os::Vocab::encode("remove"));
+
+
+    sendCmd(bCmd,true);
+}
+
+bool wysiwyd::wrdac::SubSystem_KARMA::pushLeft(const yarp::sig::Vector &objCenter, const double &targetPosYLeft,
+                                               const double &zOff,
+                                               const std::string &armType,
+                                               const yarp::os::Bottle &options, const std::string &sName)
+{
+    // Calculate the pushing distance (radius) for push with Karma
+    double radius;
+    Vector object = objCenter;
+    Bottle opt = options;
+    selectHandCorrectTarget(opt,object);    // target is calibrated by this method
+    radius = fabs(object[1] - targetPosYLeft);
+    yInfo ("objectY = %f",object[1]);
+    yInfo ("targetPosYLeft = %f",targetPosYLeft);
+    yInfo ("radius = %f",radius);
+    Vector targetCenter = object;
+    targetCenter[1] = targetPosYLeft;
+//    targetCenter[2] = std::max(zOff, targetCenter[2]);
+    if (hasTable)
+        targetCenter[2] = tableHeight + 0.1;
+    else
+        targetCenter[2] = zOff;
+    yInfo ("object height = %f",targetCenter[2]);
+
+    // Choose arm
+    bool armChoose = false;
+    if (armType =="right" || armType == "left")
+        armChoose = toolAttach(armType,Vector(3,0.0));
+
+    // Call push (no calibration)
+    bool pushSucceed = push(targetCenter,0,radius,options,sName);
+
+    if (armChoose)
+        toolRemove();
+
+    return pushSucceed;
+}
+
+bool wysiwyd::wrdac::SubSystem_KARMA::pushRight(const yarp::sig::Vector &objCenter, const double &targetPosYRight,
```

Comment:
I agree that they can merge but it will introduce the `theta` argument into the new method!